src/qt/receiverequestdialog.cpp

```diff
@@ -56,43 +49,58 @@ void ReceiveRequestDialog::setModel(WalletModel *_model)
 void ReceiveRequestDialog::setInfo(const SendCoinsRecipient &_info)
 {
     this->info = _info;
-    update();
-}
+    setWindowTitle(tr("Request payment to %1").arg(info.label.isEmpty() ? info.address : info.label));
+    QString uri = GUIUtil::formatBitcoinURI(info);
 
-void ReceiveRequestDialog::update()
-{
-    if(!model)
-        return;
-    QString target = info.label;
-    if(target.isEmpty())
-        target = info.address;
-    setWindowTitle(tr("Request payment to %1").arg(target));
+#ifdef USE_QRCODE
+    if (ui->qr_code->setQR(uri, info.address)) {
+        connect(ui->btnSaveAs, &QPushButton::clicked, ui->qr_code, &QRImageWidget::saveImage);
+    } else {
+        ui->btnSaveAs->setEnabled(false);
+    }
+#else
+    ui->btnSaveAs->hide();
+    ui->qr_code->hide();
+#endif
 
-    QString uri = GUIUtil::formatBitcoinURI(info);
-    ui->btnSaveAs->setEnabled(false);
-    QString html;
-    html += "<html>";
-    html += "<b>" + tr("Payment information") + "</b><br><br>";
-    html += "<b>"+tr("URI")+"</b>: ";
-    html += QString("<a style=\"%1\"href=\"").arg(GUIUtil::getThemedStyleQString(GUIUtil::ThemedStyle::TS_COMMAND)) +
-            uri + "\">" + GUIUtil::HtmlEscape(uri) + "</a><br>";
-    html += "<b>"+tr("Address")+"</b>: " + GUIUtil::HtmlEscape(info.address) + "<br>";
-    if(info.amount)
-        html += "<b>"+tr("Amount")+"</b>: " + BitcoinUnits::formatHtmlWithUnit(model->getOptionsModel()->getDisplayUnit(), info.amount) + "<br>";
-    if(!info.label.isEmpty())
-        html += "<b>"+tr("Label")+"</b>: " + GUIUtil::HtmlEscape(info.label) + "<br>";
-    if(!info.message.isEmpty())
-        html += "<b>"+tr("Message")+"</b>: " + GUIUtil::HtmlEscape(info.message) + "<br>";
-    if(model->isMultiwallet()) {
-        html += "<b>"+tr("Wallet")+"</b>: " + GUIUtil::HtmlEscape(model->getWalletName()) + "<br>";
+    ui->uri_content->setText("<a href=\"" + uri + "\">" + GUIUtil::HtmlEscape(uri) + "</a>");
```

Comment:
17597: we use themes, should keep applying styles here
```suggestion
    QString uri_style = GUIUtil::getThemedStyleQString(GUIUtil::ThemedStyle::TS_COMMAND);
    ui->uri_content->setText(QString("<a style=\"%1\" href=\"%2\">%3</a>").arg(uri_style).arg(uri).arg(GUIUtil::HtmlEscape(uri)));
```

Reply:
I pushed suggestion. It compiles, dialog is rendered but not sure how to check validity of style usage